pkg/auth/cli.go

```diff
@@ -45,7 +45,7 @@ type LogoutOptions struct {
 // GetLoginFlags defines and returns login flags for containers tools
 func GetLoginFlags(flags *LoginOptions) *pflag.FlagSet {
 	fs := pflag.FlagSet{}
-	fs.StringVar(&flags.AuthFile, "authfile", GetDefaultAuthFile(), "path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
+	fs.StringVar(&flags.AuthFile, "authfile", "", "path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
```

Comment:
I like to reveal this information to the user via

podman login --help 

Why are you removing this information?

Reply:
We need to detect whether the user incorrectly passed `--authfile …` `--compat-auth-file …`, and setting a default value here interferes with that.

Also, `GetDefaulutAuthFile` does not do what it suggests; on almost all systems it returns `""`.

Reply:
The idea is revealing the default, if user specifies the --authfile or --docker-auth-file, then they modified the defaults.
This should be showing $XDG_RUNTIME_DIR/UID/auth.json almost everywhere unless there is a bug.

It would also show the correct value if DOCKER_AUTH_FILE or REGISTRY_AUTH_FILE environment is set.

Reply:
But that’s not how it works.

On the read paths (not `login`/`logout`, this has always been incorrect because “no `--authfile`” and `--authfile=$the_default_from_--help`” don’t do the same thing: the former searches for `auth.json` in `/run`, and in `~/.config`, and for two formats of Docker-defined files. The latter _only_ reads that one file. (I think fairly strongly that this behavior is correct.)

And, on both read paths and login/logout, if a credential helper is configured in `registries.conf`, the value of `--authfile` (and of the new `--compat-auth-file`) might be effectively ignored. (I don’t know whether that’s desirable, it’s just what the code does.)

I see some value in displaying the paths that will be used in `--help`; but doing that by setting “the default value of the flag”, and in effect having the flag always set and non-empty, is an accurate way to do that.

Reply:
Change 

fs.StringVar(&flags.AuthFile, "authfile", GetDefaultAuthFile(), "path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")

To
fs.String("authfile", GetDefaultAuthFile(), "path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")

Then during the main function set the values only if they changed, otherwise rely on the regular processioning as if they were "".

Reply:
- That doesn’t actually implement the feature you want: `GetDefaultAuthFile` returns `""` on most systems. (and it must _continue_ to do so, given how it is currently called in Buildah/Podman, otherwise Podman would stop searching in any of the ~3 other locations.)
- And a description of the default behavior into `--help` as a default value of `--authfile` is actively confusing, as described above. The default behavior is a credential helper user and/or a file search, passing any `--authfile` value disables that search (and not the credential helper use)
- As a special case of that confusion, see how `$DOCKER_CONFIG` is handled. `GetDefaultAuthFile` must continue to read that environment variable, because Podman and Buildah “credential reader” commands rely on that and removing that behavior would be a regression, but `login`/`logout` must use that value in `--authfile`, because they need to write in the compatibility format.

I don’t object to idea of showing default locations in `--help`. But that feature mostly doesn’t exist right now; it is not the topic of this PR; and I don’t think showing the default locations (note the plural!) in the default value of `--authfile` can work.